src/mongo_odm/odm_collection.hpp

```diff
@@ -48,25 +48,28 @@ class odm_collection {
 
     ///
     /// Runs an aggregation framework pipeline against this collection, and returns the results
-    /// as
-    /// de-serialized object.
+    /// as de-serialized object.
+    /// This function is templated on the result's type, which is not necessarily the same as the
+    /// type of the documents in the colleciton.
     ///
+    /// @tparam Result - The type of the aggregation result.
     /// @param pipeline
     ///   The pipeline of aggregation operations to perform.
     /// @param options
     ///   Optional arguments, see mongocxx::mongocxx::options::aggregate.
     ///
-    /// @return A deserializing_cursor with the results.
+    /// @return A deserializing_cursor<Result> with the results.
     /// @throws
     ///   If the operation failed, the returned cursor will throw an mongocxx::exception::query
     ///   when it is iterated.
     ///
     /// @see http://docs.mongodb.org/manual/reference/command/aggregate/
     ///
-    deserializing_cursor<T> aggregate(
+    template <class Result>
```

Comment:
In case someone wants to do an aggregation that returns a cursor of `T`'s, I think we should default this here, e.g.
`template <class Result = T>`